beacon-chain/state/state-native/v1/getters_state.go

```diff
@@ -1,115 +1,188 @@
 package v1
 
 import (
+	"fmt"
+
 	"github.com/pkg/errors"
-	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
 	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
 )
 
-// InnerStateUnsafe returns the pointer value of the underlying
+// ToProtoUnsafe returns the pointer value of the underlying
 // beacon state proto object, bypassing immutability. Use with care.
-func (b *BeaconState) InnerStateUnsafe() interface{} {
+func (b *BeaconState) ToProtoUnsafe() interface{} {
 	if b == nil {
 		return nil
 	}
-	return b.state
+
+	var bRoots [][]byte
```

Comment:
You can use our getters here rather than rewriting everything

Reply:
`ToProtoUnsafe` is called from `HashTreeRoot`, which obtains the state lock. Using the getter here will result in a deadlock.